Pcb Improvements

MarcusCalhoun-Lopez opened the pull request:
#### Description

<!-- Note: it is best make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Generate version information with this command in shell:
    echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk '{print $NF}' | tr '\n' ' ')"
-->
macOS 10.14
Xcode 10.1

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

technobauble commented:
Opps, I thought it would let me review the commits one at a time.

264211d: Good.

9669120: Good.

e0b65d6: The version of gts that we include is customized. We need to insure that the our version of the header is found before any other system version. By renaming it, the problem is guaranteed solved without having to worry about the build system. Is there a compelling reason to change this? I'll take your word for it on the GL includes.

79f3aef: Okay.

9630cb4: I think that building the documentation should be enabled as a default variant. There are (unfortunately) some things in the docs that aren't well documented elsewhere.

a8bfe71: Right now the default is to use quartz. It looks to me like this would change that. Is there a compelling reason to do so?

8dbb485: Should I be concerned that the travis build is failing?

MarcusCalhoun-Lopez commented:
> [e0b65d6](https://github.com/macports/macports-ports/commit/e0b65d68a87da98342ef09fcaca04019c2df1992): The version of gts that we include is customized. We need to insure that the our version of the header is found before any other system version. By renaming it, the problem is guaranteed solved without having to worry about the build system. Is there a compelling reason to change this? I'll take your word for it on the GL includes.

I modified the pull request so that gts is again renamed.
 
> [9630cb4](https://github.com/macports/macports-ports/commit/9630cb43e61357accfe7cd206f03c907d53f8ca3): I think that building the documentation should be enabled as a default variant. There are (unfortunately) some things in the docs that aren't well documented elsewhere.

I modified the pull request so `docs` is a default variant.

> [a8bfe71](https://github.com/macports/macports-ports/commit/a8bfe71d7b48daf5444ffebf35cd327525a2f191): Right now the default is to use quartz. It looks to me like this would change that. Is there a compelling reason to do so?

Isn't the default currently x11?
[gtk2](https://github.com/macports/macports-ports/blob/1462f078504c2c48173a4d949c3948462c7f7472/gnome/gtk2/Portfile#L132), [gd2](https://github.com/macports/macports-ports/blob/1462f078504c2c48173a4d949c3948462c7f7472/graphics/gd2/Portfile#L86), and [gtkglext](https://github.com/macports/macports-ports/blob/1462f078504c2c48173a4d949c3948462c7f7472/devel/gtkglext/Portfile#L40) are all `+x11` by default.
Currently, pcb relies on [mesa](https://github.com/macports/macports-ports/blob/1462f078504c2c48173a4d949c3948462c7f7472/x11/pcb/Portfile#L40).
Please forgive me if I am missing something.

> [8dbb485](https://github.com/macports/macports-ports/commit/8dbb485aea7724f5c193f24bbf738f6565ca4b56): Should I be concerned that the travis build is failing?

I am afraid the travis build system is a mystery to me.
It has failed for me many times on changes that were correct.

technobauble commented:
> > [a8bfe71](https://github.com/macports/macports-ports/commit/a8bfe71d7b48daf5444ffebf35cd327525a2f191): Right now the default is to use quartz. It looks to me like this would change that. Is there a compelling reason to do so?
> 
> Isn't the default currently x11?
> [gtk2](https://github.com/macports/macports-ports/blob/1462f078504c2c48173a4d949c3948462c7f7472/gnome/gtk2/Portfile#L132), [gd2](https://github.com/macports/macports-ports/blob/1462f078504c2c48173a4d949c3948462c7f7472/graphics/gd2/Portfile#L86), and [gtkglext](https://github.com/macports/macports-ports/blob/1462f078504c2c48173a4d949c3948462c7f7472/devel/gtkglext/Portfile#L40) are all `+x11` by default.
> Currently, pcb relies on [mesa](https://github.com/macports/macports-ports/blob/1462f078504c2c48173a4d949c3948462c7f7472/x11/pcb/Portfile#L40).
> Please forgive me if I am missing something.

Maybe I am. I haven't explicitly built anything as a quartz variant, but when I run it, it runs as a quartz window. So that was my assumption. When I build the default version of your new Portfile, it also runs in quartz...

> > [8dbb485](https://github.com/macports/macports-ports/commit/8dbb485aea7724f5c193f24bbf738f6565ca4b56): Should I be concerned that the travis build is failing?
> 
> I am afraid the travis build system is a mystery to me.
> It has failed for me many times on changes that were correct.

Okay. 

Alright, I'll approve all of these changes now. Thanks for your contributions!
